### PR TITLE
Remove `o_` prefixes

### LINKED ISSUE #174: [Task]: Remove o_ prefix for data object properties and database columns
### Improvement description

Please see https://github.com/pimcore/pimcore/issues/7341 for details

### PR DESCRIPTION
**Pull request by aashan10**
Resolves #174

### CODE REVIEW

**Comment by aashan10** [comment]
Resolves #174

File: src/Installer.php [hunk 1]
```diff
@@ -52,12 +52,12 @@ public function install()
 
         if (! $schema->hasTable(self::QUEUE_TABLE_NAME)) {
             $queueTable = $schema->createTable(self::QUEUE_TABLE_NAME);
-            $queueTable->addColumn('o_id', 'bigint', ['default' => 0, 'notnull' => true]);
+            $queueTable->addColumn('id', 'bigint', ['default' => 0, 'notnull' => true]);
```
Comment (dvesh3) [inline src/Installer.php]: migration?

File: src/Controller/AdminController.php [hunk 2]
```diff
@@ -135,20 +137,24 @@ public function gridProxyAction(Request $request, Service $service, EventDispatc
...
+            $idField = DataObjectService::getVersionDependentDatabaseColumnName('id');
+            $keyColumn = DataObjectService::getVersionDependentDatabaseColumnName('key');
+            $pathColumn = DataObjectService::getVersionDependentDatabaseColumnName('path');
             if (!$this->getAdminUser()->isAdmin()) {
                 $userIds = $this->getAdminUser()->getRoles();
                 $userIds[] = $this->getAdminUser()->getId();
+
                 $conditionFilters[] = ' (
-                                                    (select list from users_workspaces_object where userId in (' . implode(',', $userIds) . ') and LOCATE(CONCAT(o_path,o_key),cpath)=1  ORDER BY LENGTH(cpath) DESC LIMIT 1)=1
+                                                    (select list from users_workspaces_object where userId in (' . implode(',', $userIds) . ') and LOCATE(CONCAT('. $keyColumn .', ' . $pathColumn . '),cpath)=1  ORDER BY LENGTH(cpath) DESC LIMIT 1)=1
```
Comment (dvesh3) [inline src/Controller/AdminController.php]:
```suggestion
                                                    (select list from users_workspaces_object where userId in (' . implode(',', $userIds) . ') and LOCATE(CONCAT('. $pathColumn .', ' . $keyColumn . '),cpath)=1  ORDER BY LENGTH(cpath) DESC LIMIT 1)=1
```

File: src/Controller/AdminController.php [hunk 3]
```diff
@@ -135,20 +137,24 @@ public function gridProxyAction(Request $request, Service $service, EventDispatc
...
                                                     OR
-                                                    (select list from users_workspaces_object where userId in (' . implode(',', $userIds) . ') and LOCATE(cpath,CONCAT(o_path,o_key))=1  ORDER BY LENGTH(cpath) DESC LIMIT 1)=1
+                                                    (select list from users_workspaces_object where userId in (' . implode(',', $userIds) . ') and LOCATE(cpath,CONCAT('. $keyColumn .', ' . $pathColumn . '))=1  ORDER BY LENGTH(cpath) DESC LIMIT 1)=1
```
Comment (dvesh3) [inline src/Controller/AdminController.php]:
```suggestion
                                                    (select list from users_workspaces_object where userId in (' . implode(',', $userIds) . ') and LOCATE(cpath,CONCAT('. $pathColumn .', ' . $keyColumn . '))=1  ORDER BY LENGTH(cpath) DESC LIMIT 1)=1
```

File: src/Resources/config/pimcore/config.yml [hunk 4]
```diff
@@ -51,33 +51,33 @@ advanced_object_search:
 
     core_fields_configuration:
         -
-            field: o_id
+            field: id
```
Comment (dvesh3) [inline src/Resources/config/pimcore/config.yml:54]: how this suppose to work on Pimcore 10 with `o_` prefix?

File: src/Service.php [hunk 5]
```diff
@@ -519,11 +519,11 @@ protected function updateUpdateQueueForDataObject(Concrete $object)
         $db = \Pimcore\Db::get();
 
         //add object to update queue (if not exists) or set in_queue to false
-        $currentEntry = $db->fetchAssociative('SELECT in_queue FROM ' . Installer::QUEUE_TABLE_NAME . ' WHERE o_id = ?', [$object->getId()]);
+        $currentEntry = $db->fetchAssociative('SELECT in_queue FROM ' . Installer::QUEUE_TABLE_NAME . ' WHERE id = ?', [$object->getId()]);
         if (!$currentEntry) {
-            $db->insert(Installer::QUEUE_TABLE_NAME, ['o_id' => $object->getId(), 'classId' => $object->getClassId()]);
+            $db->insert(Installer::QUEUE_TABLE_NAME, ['id' => $object->getId(), 'classId' => $object->getClassId()]);
```
Comment (dvesh3) [inline src/Service.php]: why not `getVersionDependentDatabaseColumnName` here? and for other changes in this service.

File: src/Resources/config/pimcore/config.yml (hunk 4 above)
Comment (dvesh3) [reply src/Resources/config/pimcore/config.yml:54]: as discussed, this should be fine as the value is based on return value from Service

**Pull request merged**